infra/blueprint-test/pkg/tft/terraform.go

```diff
@@ -272,6 +272,37 @@ func NewTFBlueprintTest(t testing.TB, opts ...tftOption) *TFBlueprintTest {
 	return tft
 }
 
+// TFBlueprintTestForNonTest encapsulates the necessary parts of TFBlueprintTest
+// for use outside of a testing context.
+type TFBlueprintTestForNonTest struct {
+	*TFBlueprintTest
+	Assertions *assert.Assertions // Add an Assertions field
+}
+
+// NewTFBlueprintTestForNonTest creates a TFBlueprintTest instance suitable for
+// non-testing scenarios.
+func NewTFBlueprintTestForNonTest(opts ...tftOption) (*TFBlueprintTestForNonTest, error) {
```

Comment:
I am not sure I follow why we need this, could we use `go-testing-interface` for mocking out T?

https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/blueprint-test/pkg/tft/terraform_test.go#L70

We also have some examples in https://github.com/terraform-google-modules/terraform-example-foundation/blob/0bc707bc665d31ca9d20f8e984e780bf656d99e4/helpers/foundation-deployer/stages/apply.go